src/iop/censorize.c

```diff
@@ -0,0 +1,455 @@
+{
+  return IOP_FLAGS_INCLUDE_IN_STYLES | IOP_FLAGS_SUPPORTS_BLENDING | IOP_FLAGS_ALLOW_TILING;
+}
+
+int default_group()
+{
+  return IOP_GROUP_EFFECT | IOP_GROUP_EFFECTS;
+}
+
+int default_colorspace(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece)
+{
+  return iop_cs_rgb;
+}
+
+void tiling_callback(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t *piece,
+                     const dt_iop_roi_t *roi_in, const dt_iop_roi_t *roi_out,
+                     struct dt_develop_tiling_t *tiling)
+{
+  dt_iop_censorize_data_t *d = (dt_iop_censorize_data_t *)piece->data;
+
+  const float radius_1 = fmax(0.1f, d->radius_1);
+  const float sigma = radius_1 * roi_in->scale / piece->iscale;
+
+  const int width = roi_in->width;
+  const int height = roi_in->height;
+  const int channels = piece->colors;
+
+  const size_t basebuffer = width * height * channels * sizeof(float);
+  tiling->factor = 2.0f + fmax(1.0f, (float)dt_gaussian_memory_use(width, height, channels) / basebuffer);
+  tiling->maxbuf = fmax(1.0f, (float)dt_gaussian_singlebuffer_size(width, height, channels) / basebuffer);
+  tiling->overhead = 0;
+  tiling->overlap = ceilf(4 * sigma);
+  tiling->xalign = 1;
+  tiling->yalign = 1;
+  return;
+}
+
+static inline void make_noise(float *const output, const float noise, const size_t width, const size_t height)
+{
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
+  dt_omp_firstprivate(output, width, height, noise) \
+  schedule(simd:static) aligned(output:64) collapse(2)
+#endif
+  for(size_t i = 0; i < height; i++)
+    for(size_t j = 0; j < width; j++)
+    {
+      // Init random number generator
+      uint32_t DT_ALIGNED_ARRAY state[4] = { splitmix32(j + 1), splitmix32((j + 1) * (i + 3)), splitmix32(1337), splitmix32(666) };
+      xoshiro128plus(state);
+      xoshiro128plus(state);
+      xoshiro128plus(state);
+      xoshiro128plus(state);
+
+      const size_t index = (i * width + j) * 4;
+      float *const restrict pix_out = __builtin_assume_aligned(output + index, 16);
+      const float norm = pix_out[1];
+
+      // create statistical noise
+      const float epsilon = gaussian_noise(norm, noise * norm, i % 2 || j % 2, state) / norm;
+
+      // add noise to output
+      for(size_t c = 0; c < 3; c++) pix_out[c] *= epsilon;
+    }
+}
+
+
+void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *const ivoid,
+             void *const ovoid, const dt_iop_roi_t *const roi_in, const dt_iop_roi_t *const roi_out)
+{
+  dt_iop_censorize_data_t *data = (dt_iop_censorize_data_t *)piece->data;
+  const float *const restrict in = DT_IS_ALIGNED((const float *const restrict)ivoid);
+  float *const restrict out = DT_IS_ALIGNED((float *const restrict)ovoid);
+
+  const int width = roi_in->width;
+  const int height = roi_in->height;
+  const int ch = 4;
+  assert(piece->colors == ch);
+
+  float *const restrict temp = dt_alloc_sse_ps(width * height * ch);
```

Comment:
dt_alloc_sse_ps() -> dt_alloc_align_float()

Same def, but better to use later in non SSE code path.

Reply:
Will do.